pyspeckit/spectrum/units.py

```diff
@@ -394,13 +394,21 @@ def units(self, value):
     @property
     def refX_units(self):
         log.warn("'refX_units' is deprecated; please use 'refX_unit'", DeprecationWarning)
-        return self.refX_unit
+        return self.refX.unit
 
     @refX_units.setter
     def refX_units(self, value):
         log.warn("'refX_units' is deprecated; please use 'refX_unit'", DeprecationWarning)
         self.refX_unit = value
 
+    @property
+    def refX_unit(self):
+        return self.refX.unit
+
+    @refX_unit.setter
+    def refX_unit(self, value):
+        log.warn("'refX_unit' should not be set directly, instead modify refX")
```

Comment:
hmm, yes I'm trying to find a solution for this, __array_finalize__ is in the way so there has to be a way refX_unit can be set.